### PR TITLE
abs

### PR DESCRIPTION
**Pull request by marcelbischoff**
did I got it wrong the first time?

### CODE REVIEW

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def leakyrelu(self, neg_slope=0.01):
     return self.relu() + (-neg_slope*self).relu()
 
   def abs(self):
-    return self.relu() + (-1.0*self).relu()*(-1.0)
+    return self.relu() + (-1.0*self).relu()
```
Comment (kartik4949) [inline tinygrad/tensor.py:207]: def leakyrelu(self, neg_slope=0.01):
    return self.relu() + (-neg_slope*self).relu()

isnt the above leakyrelu with neg_slope = -1.0

Comment (kartik4949) [reply tinygrad/tensor.py:207]: why not use it as 
def abs(self):
   return self.leakyrelu(1.0)

**Comment by geohot** [comment]
Wait what? Can you improve the test?

File: tinygrad/tensor.py [hunk 2]
```diff
@@ -204,7 +204,7 @@ def leakyrelu(self, neg_slope=0.01):
     return self.relu() + (-neg_slope*self).relu()
```
Comment (geohot) [inline tinygrad/tensor.py:204]: I don't think this is right.

Assume self = -1
(-0.1*self).relu() = 0.1

You want 
```
self.relu() + (-neg_slope*self).relu()*(-1.0)
```

And abs is
```
self.relu() + (-1.0*self).relu()
```

Please improve the tests for this

**Comment by marcelbischoff** [comment]
Yes, they were both wrong. Tests are positive so not caught.

**Comment by geohot** [comment]
We need negative tests.

**Comment by kartik4949** [comment]
> We need negative tests. For all the tests

uhm range change  -1 to 0 for random values may be

File: tinygrad/tensor.py (hunk 2 above)
Comment (kartik4949) [reply tinygrad/tensor.py:204]: self.relu - (-neg_slope*self).relu()

**Pull request merged**